os/arch/arm/src/armv8-m/up_hardfault.c

```diff
@@ -83,7 +87,11 @@
  */
 
 #ifdef CONFIG_DEBUG_HARDFAULT
+#ifdef CONFIG_SECURITY_LEVEL
+#define hfdbg(format, ...) if (get_security_level() == LOW_SECURITY_LEVEL) lldbg(format, ##__VA_ARGS__)
+#else
 #define hfdbg(format, ...) lldbg(format, ##__VA_ARGS__)
+#endif
```

Comment:
There is no difference between ifdef and else. 
Why do we need to seperate the dbg according to the CONFIG?